infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/aws/AWSResourceManager.java

```diff
@@ -0,0 +1,297 @@
+ */
+public class AWSResourceManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(AWSResourceManager.class);
+
+    private AWSResourceLimitUOW awsResourceUOW;
+    private static final String EC2_RESOURCE_IDENTIFIER = "AWS::EC2::Instance";
+    private static final String VPC_RESOURCE_IDENTIFIER = "AWS::EC2::VPC";
+    private static final String EC2_SERVICE_NAME = "ec2";
+    private static final String NETWORKING_SERVICE_NAME = "networking";
+    private static final String VPC_LIMIT_NAME = "vpc";
+
+    public AWSResourceManager() {
+        awsResourceUOW = new AWSResourceLimitUOW();
+    }
+
+    /**
+     * Parses the limits.yaml and returns the AWS resources and limits.
+     * This is used to populate the initial AWSResourceLimits
+     *
+     * @param limitFilePath path to limits.yaml file
+     * @return AWSResourceLimits instance
+     * @throws IOException if reading limits.yaml fails
+     * @throws TestGridInfrastructureException if error occurs while retrieving AWS resource limits
+     */
+    public AWSResourceLimit getInitialResourceLimits(Path limitFilePath) throws TestGridInfrastructureException {
+        String yamlContent;
+        try {
+            yamlContent = new String(Files.readAllBytes(limitFilePath), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new TestGridRuntimeException("Error while reading yaml content.");
+        }
+        Representer representer = new Representer();
+
+        // Skip missing properties in testgrid.yaml
+        representer.getPropertyUtils().setSkipMissingProperties(true);
+        AWSResourceLimit awsResourceLimits = new Yaml(new Constructor(AWSResourceLimit.class), representer)
+                .loadAs(yamlContent, AWSResourceLimit.class);
+        if (awsResourceLimits == null) {
+            throw new TestGridInfrastructureException(
+                    "Error occurred while parsing" + limitFilePath.toString());
+        }
+        return awsResourceLimits;
+    }
+
+    /**
+     * Retrieves and available region to run a test plan.
+     *
+     * @param resourceRequirements resources requirements of a test plan
+     * @return available region
+     * @throws TestGridDAOException if retrieving information from the database failed
+     */
+    public String getAvailableRegion(List<DeploymentPatternResourceUsage> resourceRequirements)
+            throws TestGridDAOException {
+
+        if (!awsResourceUOW.getAWSRegions().isPresent()) {
+            throw new TestGridDAOException("No AWS region information found in the database");
+        }
+        List<String> regions = awsResourceUOW.getAWSRegions().get();
+        for (String region : regions) {
+            boolean isRegionAvailable = true;
+            for (DeploymentPatternResourceUsage resourceRequirement : resourceRequirements) {
+                AWSResourceLimit awsResourceLimit = awsResourceUOW.getAvailableResource(resourceRequirement, region);
```

Comment:
We can define AWSResourceLimit object before the for-loop and avoid creating object-per loop turn.